src/main/java/im/status/keycard/KeycardApplet.java

```diff
@@ -847,6 +861,49 @@ private void loadSeed(byte[] apduBuffer) {
     resetKeyStatus();
     JCSystem.commitTransaction();
   }
+  
+  // Load up to 155 bytes of data into a buffer. This will generally be used for Shamir secrets or
+  // Bitcoin multisig scripts. The buffer (`savedData`) does not interact with anything in the applet.
+  // [PIN protected APDU]
+  private void loadData(APDU apdu) {
+    byte[] apduBuffer = apdu.getBuffer();
+    secureChannel.preprocessAPDU(apduBuffer);
+
+    if (!pin.isValidated()) {
+      ISOException.throwIt(ISO7816.SW_CONDITIONS_NOT_SATISFIED);
+    }
+    
+    if (apduBuffer[ISO7816.OFFSET_LC] > SAVED_DATA_SIZE) {
+      ISOException.throwIt(ISO7816.SW_WRONG_DATA);
+    }
+    
+    JCSystem.beginTransaction();
+    Util.arrayCopy(apduBuffer, (short) ISO7816.OFFSET_CDATA, savedData, (short) 0, apduBuffer[ISO7816.OFFSET_LC]);
+    JCSystem.commitTransaction();
+    
+    secureChannel.respond(apdu, (short) 0, ISO7816.SW_NO_ERROR);
+  }
+  
+  // Export the entirety of the `savedData` buffer (155 bytes).
+  // [PIN protected APDU]
+  private void exportData(APDU apdu) {
+    byte[] apduBuffer = apdu.getBuffer();
+    secureChannel.preprocessAPDU(apduBuffer);
+
+    if (!pin.isValidated()) {
+      ISOException.throwIt(ISO7816.SW_CONDITIONS_NOT_SATISFIED);
+    }
+
+    short off = SecureChannel.SC_OUT_OFFSET;
+    apduBuffer[off++] = TLV_DATA;
+    apduBuffer[off++] = (byte) SAVED_DATA_SIZE;
```

Comment:
There is no need to TLV encode this, so lets not add the extra TLV overhead. If an application decides to store data in this "data hole", they can choose to encode that data as desired.

Reply:
As long as it doesn't break our interface, I don't care, but are we concerned about the API break relative to all other APDUs?

Reply:
Several APDUs do not have TLV encoded payloads (mutual auth comes to mind first). The distinction seems to be drawn at how many fields are in the payload. If the payload is just one byte array, then no TLV encoding is added. If the payload consists of several protocol defined fields, then the fields are often separated and encoded TLV.

Sooooo short answer is... since this is just an amorphous byte array (from the protocol's view), then it should not be TLV encoded

Reply:
998c77d